db-engine/src/main/java/com/splicemachine/db/impl/sql/GenericStatement.java

```diff
@@ -768,6 +772,15 @@ private void setTimestampFormat(LanguageConnectionContext lcc, CompilerContext c
         }
     }
 
+    private void setCursorUntypedExpressionType(LanguageConnectionContext lcc, CompilerContext cc) throws StandardException {
+        String type = PropertyUtil.getCachedDatabaseProperty(lcc, Property.CURSOR_UNTYPED_EXPRESSION_TYPE);
+        if(type != null && "varchar".equals(type.toLowerCase())) {
+            cc.setCursorUntypedExpressionType(DataTypeDescriptor.getBuiltInDataTypeDescriptor(Types.VARCHAR, true, 254));
+        } else {
+            cc.setCursorUntypedExpressionType(null);
```

Comment:
Instead of silently setting it to `null`, would it make more sense to throw e.g. UnsupportedTypeException or something similar?

Reply:
Will do, thanks!

Reply:
Tracked by DB-11617